go/serialization.go

```diff
@@ -350,31 +350,59 @@ func SerializeSCTSignatureInput(sct SignedCertificateTimestamp, entry LogEntry)
 	}
 }
 
-func serializeV1SCT(sct SignedCertificateTimestamp) ([]byte, error) {
-	if err := checkExtensionsFormat(sct.Extensions); err != nil {
-		return nil, err
-	}
-	var buf bytes.Buffer
-	if err := binary.Write(&buf, binary.BigEndian, V1); err != nil {
-		return nil, err
-	}
-	if err := binary.Write(&buf, binary.BigEndian, sct.LogID); err != nil {
-		return nil, err
+func serializeV1SCTHere(sct SignedCertificateTimestamp, here []byte) (int, error) {
+	if sct.SCTVersion != 0 {
+		return 0, INVALID_VERSION
 	}
-	if err := binary.Write(&buf, binary.BigEndian, sct.Timestamp); err != nil {
-		return nil, err
+
+	//Determine Lenth of Serialized Output
+	ctxlen := len(sct.Extensions)
+	siglen := len(sct.Signature.Signature)
+	sctoutput := 1 + 32 + 8 + 2 + ctxlen + 2 + 2 + siglen
+
+	if sctoutput > 0xFFFF {
+		return 0, SCT_TOO_LARGE
 	}
-	if err := writeVarBytes(&buf, sct.Extensions, ExtensionsLengthBytes); err != nil {
-		return nil, err
+	if len(here) < sctoutput {
+		return sctoutput, NOT_ENOUGH_BUFFER
 	}
-	sig, err := MarshalDigitallySigned(sct.Signature)
-	if err != nil {
-		return nil, err
-	}
-	if err := binary.Write(&buf, binary.BigEndian, sig); err != nil {
+
+	//Write Version
+	here[0] = byte(sct.SCTVersion)
+
+	//Write LogID
+	copy(here[1:33], sct.LogID[:])
+
+	//Write Timestamp
+	binary.BigEndian.PutUint64(here[33:41], sct.Timestamp)
+
+	//Write Extensions
+	binary.BigEndian.PutUint16(here[41:43], uint16(ctxlen))
```

Comment:
You've removed the call to `checkExtensionsFormat()`, could you add it back in?
(I get that in effect it's covered by the check for `sctoutput > 0xFFFF` above, but if those format checks become more involved at some point in the future we'll have to remember to search around for 'inlined' serialisations like this one, increasing the chances of introducing bugs/inconsistencies.